### PR TITLE
Add GCP Secret Manager

### PR DESCRIPTION
**Pull request by dev-drprasad**
# Description

Add GCP Secret Manager as secret store

## Issue reference

this PR will close: #198 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by dev-drprasad** [comment]
Hi @yaron2 @youngbupark ,

GCP secret manager client doesn't contain any interface. How can i test without interface ?

**Comment by dev-drprasad** [comment]
Even though i used const  variable `versionID` in code, linter throwing an error
```
Run make lint
# Due to https://github.com/golangci/golangci-lint/issues/580, we need to add --fix for windows
golangci-lint run --fix
secretstores/gcp/secretmanager/secretmanager.go:14:7  unused  const `versionID` is unused
secretstores/gcp/secretmanager/secretmanager.go:33:6  golint  type name will be used as secretmanager.SecretManagerStore by other packages, and that stutters; consider calling this Store
```

**Comment by yaron2** [comment]
Great. Please see my comment above

**Comment by dev-drprasad** [comment]
@yaron2 I don't see any comments

**Comment by yaron2** [comment]
Oops forgot to submit. See now.

File: secretstores/gcp/secretmanager/secretmanager.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	ctx := context.Background()
+	accessRequest := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: fmt.Sprintf("projects/%s/secrets/%s/versions/%s", sm.ProjectID, req.Name, versionID),
+	}
+	result, err := sm.client.AccessSecretVersion(ctx, accessRequest)
+	if err != nil {
+		return res, fmt.Errorf("failed to access secret version: %v", err)
+	}
+	return secretstores.GetSecretResponse{Data: map[string]string{secretstores.DefaultSecretRefKeyName: string(result.Payload.Data)}}, nil
```
[the remark this answers is not in the document]
Comment (dev-drprasad) [reply secretstores/gcp/secretmanager/secretmanager.go]: Makes sense

**Comment by dev-drprasad** [comment]
@yaron2 

There are two things i would like to mention

1. I didn't write test for working scenario as i am not sure how to mock gcp client. But i tested locally with original gcp account
2. I haven't removed code related to cloudkms

**Comment by yaron2** [comment]
If you tested with a GCP account that's good.

No need to remove kms in this PR, that can be done with a dedicated PR.

**Comment by yaron2** [comment]
Can you take this off of Draft mode? Are you ready to merge this?

**Comment by dev-drprasad** [comment]
metadata parameters are expected in snake_case. I see few other stores use cameCase. Is that okay ? or do we need to make it consistent. ?

**Comment by yaron2** [comment]
No, gcp uses snake case credentials

**Comment by dev-drprasad** [comment]
We are good to merge then

**Comment by yaron2** [comment]
@dev-drprasad can you run `go mod tidy` and push the changes?

**Comment by dev-drprasad** [comment]
@yaron2 i ran `go mod tidy` and pushed the changes

**Pull request merged**